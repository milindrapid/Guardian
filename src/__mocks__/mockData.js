export const obj = {
  "sidebar-0": {
    name: "home",
  },
  "sidebar-1": {
    name: "about",
  },
};

export const arrayItems = [
  { children: null, id: "sidebar-0", name: "home" },
  { children: null, id: "sidebar-1", name: "about" },
];
