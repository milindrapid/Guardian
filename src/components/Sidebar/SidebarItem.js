import React, { useState } from "react";
import PropTypes from "prop-types";
import "./Sidebar.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleUp, faAngleDown } from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-router-dom";

const SidebarItem = (props) => {
  const [isExpanded, setIsEcpanded] = useState(props.isExpanded);
  const label = props.name;
  const items = props.children;
  const to = props.to;
  const depthStep = 10;
  const depth = 0;
  return (
    <>
      <li>
        <div
          className="sidebar__item"
          style={{ paddingLeft: depth * depthStep }}
        >
          <Link to={to}>
            <span>{label}</span>
          </Link>
          {Array.isArray(items) && (
            <div
              className="sidebar__expand-btn"
              onClick={() => setIsEcpanded(!isExpanded)}
            >
              <FontAwesomeIcon
                icon={isExpanded ? faAngleUp : faAngleDown}
                className={isExpanded ? "active" : ""}
              />
            </div>
          )}
        </div>
        {isExpanded && (
          <>
            {Array.isArray(items) ? (
              <ul>
                {items.map((subItem) => {
                  return (
                    <SidebarItem
                      key={subItem.name}
                      depth={depth + 1}
                      depthStep={depthStep}
                      isExpanded={subItem.is_expanded}
                      {...subItem}
                    />
                  );
                })}
              </ul>
            ) : null}
          </>
        )}
      </li>
    </>
  );
};

SidebarItem.propTypes = {
  isExpanded: PropTypes.bool,
  name: PropTypes.string,
  to: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.object),
};

SidebarItem.defaultProps = {
  isExpanded: false,
  name: "",
  to: "/",
  children: [],
};

export default SidebarItem;
