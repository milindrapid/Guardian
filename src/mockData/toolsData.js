const toolsData = [
  {
    isFavorite: true,
    title: "DI Status Training",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec posuere nisl a placerat gravida.",
    callToAction: "",
    callToMore: "",
  },
  {
    isFavorite: false,
    title: "Living Balance Sheet",
    description:
      "Innovative thinking & advanced web technology to help you achieve financial balance",
    callToAction: "",
    callToMore: "",
  },
  {
    isFavorite: true,
    title: "eSignature App",
    description: "Innovative thinking & advanced web technology",
    callToAction: "",
    callToMore: "",
  },
  {
    isFavorite: false,
    title: "Application & Forms: iPipeine",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
    callToAction: "",
    callToMore: "",
  },
];

export default toolsData;
