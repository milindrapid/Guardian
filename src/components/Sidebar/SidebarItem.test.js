import React from "react";
import { shallow } from "enzyme";
import SidebarItem from "./SidebarItem";
import { arrayItems } from "../../__mocks__/mockData";

describe("SidebarItem Component", () => {
  const props = {
    name: "home",
    children: arrayItems,
    isExpanded: false,
  };

  const wrapper = shallow(<SidebarItem {...props} />);

  it("SidebarItem Component should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render children correctly", () => {
    expect(wrapper.find("ul")).toHaveLength(0);
    expect(wrapper.find("li")).toHaveLength(1);
    expect(wrapper.find("div")).toHaveLength(2);
  });
});
