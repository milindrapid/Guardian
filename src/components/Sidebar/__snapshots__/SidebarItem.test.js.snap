// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem Component SidebarItem Component should render correctly 1`] = `
<Fragment>
  <li>
    <div
      className="sidebar__item"
      style={
        Object {
          "paddingLeft": 0,
        }
      }
    >
      <Link
        to="/"
      >
        <span>
          home
        </span>
      </Link>
      <div
        className="sidebar__expand-btn"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f107",
                "M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z",
              ],
              "iconName": "angle-down",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
    </div>
  </li>
</Fragment>
`;
